python/mxnet/io.py

```diff
@@ -455,26 +466,94 @@ def _init_data(data, allow_empty, default_name):
     return list(data.items())
 
 class NDArrayIter(DataIter):
-    """Iterating on either ``mx.nd.NDArray`` or ``numpy.ndarray``.
+    """Returns an iterator for ``mx.nd.NDArray`` or ``numpy.ndarray``.
+
+    Example usage:
+    ----------
+    >>> data = np.arange(40).reshape((10,2,2))
+    >>> labels = np.ones([10, 1])
+    >>> dataiter = mx.io.NDArrayIter(data, labels, 3, True, last_batch_handle='discard')
+    >>> dataiter
+    <mxnet.io.NDArrayIter object at 0x10bb2fd90>
```

Comment:
we generally don't need to show object pointers like this. Users know what they are because they are just constructed in previous line.